core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -88,7 +90,7 @@ object Broker {
                 (ep.protocolType, ep)
               }.toMap
             }
-          val rack: Option[String] = brokerInfo.get("rack").asInstanceOf[Option[String]]
+          val rack = brokerInfo.get("rack").filter(_ != null).map(_.asInstanceOf[String])
```

Comment:
Are we sure we never have a null entry for `rack`? If so, then we don't need the filter. If we sometimes store `null` in the `rack` field, then we need what I did here.